llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -27056,21 +27056,35 @@ AArch64TargetLowering::shouldExpandAtomicLoadInIR(LoadInst *LI) const {
                              : AtomicExpansionKind::LLSC;
 }
 
+// Return true if the atomic operation expansion will lower to use a library
+// call, and is thus ineligible to use an LLSC expansion.
+static bool rmwOpMayLowerToLibcall(const AtomicRMWInst *RMW) {
+  if (!RMW->isFloatingPointOperation())
+    return false;
+  switch (RMW->getType()->getScalarType()->getTypeID()) {
+  case Type::FloatTyID:
+  case Type::DoubleTyID:
+  case Type::HalfTyID:
+  case Type::BFloatTyID:
+    return false;
```

Comment:
We need to check if we're in softfp mode (`Subtarget->hasFPARMv8()`).

Reply:
Seems to not do anything?

Reply:
aarch64 has fp enabled by default; you need something like `-mattr=-fp-armv8` to test.

Reply:
That is in the test (in the parent #103701)